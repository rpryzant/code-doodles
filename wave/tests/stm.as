	mov	r0,#100
	mov	r1,#101
;	add	r1,r1,#0x13000000
	mov	r2,#102
	stm	r1,#7
	swi	#SysPutNum
	mov	r0,#'\n
	swi	#SysPutChar
	mov	r0,r1
	swi	#SysPutNum
	mov	r0,#'\n
	swi	#SysPutChar
	mov	r0,r2
	swi	#SysPutNum
	mov	r0,#'\n
	swi	#SysPutChar
	ldr	r0,97
	swi	#SysPutNum
	mov	r0,#'\n
	swi	#SysPutChar
	ldr	r0,98
	swi	#SysPutNum
	mov	r0,#'\n
	swi	#SysPutChar
	ldr	r0,99
	swi	#SysPutNum	
	mov	r0,#'\n
	swi	#SysPutChar
	ldr	r0,100
	swi	#SysPutNum	
	mov	r0,#'\n
	swi	#SysPutChar
	ldr	r0,101
	swi	#SysPutNum
	mov	r0,#'\n
	swi	#SysPutChar
	swi 	#SysHalt

