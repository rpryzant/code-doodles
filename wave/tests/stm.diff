*** tests/stm.out	2013-12-05 19:05:13.000000000 -0500
--- ./wave4.out	2013-12-05 19:05:13.000000000 -0500
***************
*** 1,8 ****
  100
  98
  102
- 0
- 100
- 101
- 102
- 0
--- 1,3 ----
