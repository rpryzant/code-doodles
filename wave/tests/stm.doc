Basic tests of store and load multiple.
