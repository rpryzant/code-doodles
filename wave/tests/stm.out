100
98
102
0
100
101
102
0
